### PR TITLE
reference: update the description about the `tidb_enable_table_partition` variable

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What is changed, added or deleted? (Required)

Update the description about the `tidb_enable_table_partition` variable.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 1]
```diff
@@ -341,11 +341,11 @@ set @@global.tidb_distsql_scan_concurrency = 10;
 
 ### tidb_enable_table_partition
 
-作用域：SESSION
+作用域：SESSION | GLOBAL
 
 默认值："auto"
 
-这个变量用来设置是否开启 TABLE PARTITION 特性。默认值 `auto` 表示开启 range partition 和 hash partion。`off` 表示关闭 TABLE PARTITION 的特性，此时语法还是会依旧兼容，只是建立的 partition table 实际上并不是真正的 partition table，而是和普通的 table 一样。`on` 表示开启，目前的作用和 `auto` 一样。
+这个变量用来设置是否开启 TABLE PARTITION 特性。默认值 `auto` 表示对于 TiDB 当前已实现了的分区表类型开启，目前 range partition 和 hash partion 以及 range column 单列的场景会生效。`off` 表示关闭 TABLE PARTITION 的特性，此时语法还是会依旧兼容，只是建立的 partition table 实际上并不是真正的 partition table，而是和普通的 table 一样。`on` 表示开启，无论功能实现与否。对于尚不支持的分区表类型，后续操作的行为不可预期，主要用于开发人员调试。
```
Comment (jackysp) [inline reference/configuration/tidb-server/tidb-specific-variables.md:348]: I'm a little confused about the `on` behavior.

Comment (yikeke) [inline reference/configuration/tidb-server/tidb-specific-variables.md:348]:
```suggestion
这个变量用来设置是否开启 TABLE PARTITION 特性。目前变量支持以下三种值：

- 默认值 `auto` 表示开启 TiDB 当前已实现了的分区表类型，目前在 range partition、hash partition 以及 range column 单列的场景会生效。
- `off` 表示关闭 TABLE PARTITION 的特性，此时语法还是会依旧兼容，只是建立的 partition table 实际上并不是真正的 partition table，而是和普通的 table 一样。
- `on` 表示开启，无论功能实现与否。对于尚不支持的分区表类型，后续操作的行为不可预期，主要用于开发人员调试。
```
1. 优化了其中一些表述
2. 由于段落有点长，建议优化一下格式

**Comment by tiancaiamao** [comment]
The code changes.
I'll update this document after https://github.com/pingcap/docs-cn/pull/2718/

**Comment by yikeke** [comment]
Please help follow up this PR, thx~ @ran-huang

**Comment by tiancaiamao** [comment]
Done @yikeke @jackysp

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 2]
```diff
@@ -341,11 +341,15 @@ set @@global.tidb_distsql_scan_concurrency = 10;
...
-作用域：SESSION
+作用域：SESSION | GLOBAL
+
+默认值："on"
 
-默认值："auto"
+这个变量用来设置是否开启 TABLE PARTITION 特性。目前变量支持以下三种值：
 
-这个变量用来设置是否开启 TABLE PARTITION 特性。默认值 `auto` 表示开启 range partition 和 hash partion。`off` 表示关闭 TABLE PARTITION 的特性，此时语法还是会依旧兼容，只是建立的 partition table 实际上并不是真正的 partition table，而是和普通的 table 一样。`on` 表示开启，目前的作用和 `auto` 一样。
+- 默认值 `on` 表示对于 TiDB 当前已实现了的分区表类型开启，目前 range partition 和 hash partion 以及 range column 单列的场景会生效。
```
Comment (yikeke) [inline reference/configuration/tidb-server/tidb-specific-variables.md:350]:
```suggestion
- 默认值 `on` 表示开启 TiDB 当前已实现了的分区表类型，目前 range partition、hash partition 以及 range column 单列的场景会生效。
```

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 3]
```diff
@@ -341,11 +341,15 @@ set @@global.tidb_distsql_scan_concurrency = 10;
...
+- `auto` 目前作用和 `on` 一样。
+- `off` 表示关闭 TABLE PARTITION 的特性，此时语法还是会依旧兼容，只是创建的表并不是真正的分区表，而是普通的表。
```
Comment (yikeke) [inline reference/configuration/tidb-server/tidb-specific-variables.md:352]:
```suggestion
- `off` 表示关闭 TABLE PARTITION 的特性，此时语法还是保持兼容，只是创建的表并不是真正的分区表，而是普通的表。
```

**Pull request merged**